internal/sip/sip.go

```diff
@@ -126,3 +127,27 @@ func (s SIP) IsAIP() bool {
 func (s SIP) IsSIP() bool {
 	return s.Type == enums.SIPTypeBornDigitalSIP || s.Type == enums.SIPTypeDigitizedSIP
 }
+
+func (s SIP) HasValidName() bool {
+	yyyymmdd := "(\\d{4})(\\d{2})(\\d{2})"
+	alphaNum := "[a-zA-Z0-9]"
+	alphaNumAndUnderscore := "[a-zA-Z0-9_]"
+
+	if s.Type == enums.SIPTypeBornDigitalSIP {
+		match, _ := regexp.MatchString(
+			fmt.Sprintf("^SIP_%s_%s+[_]?%s+$", yyyymmdd, alphaNum, alphaNumAndUnderscore),
```

Comment:
I think the square brackets in the `[_]?` part of this regex are unnecessary, as `_?` would be equivalent. I also think that the final `%s+$` is misleading as it implies that the refno must match one or more times (it's matching names with no refno because the third underscore is optional which allows it to match on the final letters of the office).

I'd think the following regex would be more readable:

`fmt.Sprintf("^SIP_%s_%s+(_%s+)?$", yyyymmdd, alphaNum, alphaNumAndUnderscore),`

See this example: https://regex101.com/r/2G8G8k/1

Reply:
Ah, nice! Good call!